src/ILLink.Tasks/LinkTask.cs

```diff
@@ -132,8 +138,13 @@ protected override string GenerateCommandLineCommands ()
 			}
 
 			HashSet<string> directories = new HashSet<string> ();
+			HashSet<string> assemblyNames = new HashSet<string> ();
```

Comment:
Actually, OrdinalIgnoreCase is recommended for files and directories, and anything symbolic. See https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings

Reply:
I think this makes sense for assemblies, since it sounds to me like the runtime's binding is case-sensitive. For directories, I think the case could matter on unix, though this should be rare. I've used the case-insensitive comparison for both in the latest change, but I'm still wondering if it's worth using a case-sensitive comparison for directories. The worst this would do is pass the same directory twice (with different casing) to the linker on windows. Thoughts?

Reply:
Why pass the same directory twice if we can easily filter them out to unique entries? :)

Reply:
We could definitely do that, though I prefer to keep platform-specific logic to a minimum. I'll stick with your recommendation above for now, and we can add back a case-sensitive directory comparison if it becomes an issue. Thanks!